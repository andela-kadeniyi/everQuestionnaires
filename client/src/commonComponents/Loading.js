import React from 'react';

const Loading = () => (
    <div className="loadingScreen">
      ...Loading
    </div>
);

export default Loading;