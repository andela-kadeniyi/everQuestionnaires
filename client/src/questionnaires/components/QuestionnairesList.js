import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Questionnaire from './Questionnaire';

const QuestionnairesList = ({ questionnaires }) => (
  <ul>
    { _.map(questionnaires, questionnaire => 
      <Questionnaire
        key={questionnaire.id}
        {...questionnaire}
      />
    )}
  </ul>
);

QuestionnairesList.propTypes = {
  questionnaires: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired).isRequired
};

export default QuestionnairesList;