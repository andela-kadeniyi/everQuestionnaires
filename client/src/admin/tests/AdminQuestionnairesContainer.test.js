import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import ConnectedAdminQuestionnairesContainer, { AdminQuestionnairesContainer } from '../containers/AdminQuestionnairesContainer';
import QuestionnairesList from '../../questionnaires/components/QuestionnairesList';
import Loading from '../../commonComponents/Loading';
import { formattedData } from './testData';

const middlewares = []
const mockStore = configureMockStore(middlewares);


// setup to return QuestionnairesContainer not the one wrapped by connected()
// this so to test just the rendering of the component
const setup = (formattedData, isLoading) => {
  const container = shallow(
      <AdminQuestionnairesContainer 
        questionnaires={formattedData} 
        isLoading={isLoading}/>
  );
    
  return {
    container
  };
}

const connectedSetup = (store) => {
  const connectedContainer = shallow(
      <ConnectedAdminQuestionnairesContainer store={store}/>
  );
  return {
    connectedContainer
  };
}

describe('AdminQuestionnaires container', () => {
  it('renders without crashing', () => {  
    const store = mockStore({
      adminQuestionnaires: {
        isLoading: false, 
        data: null,
        errorMessage: null
      },
    });
    
    const fetchAdminQuestionnairesSpy = jest.fn();
    // using mount for Full DOM rendering to make sure 
    //the component doesn't crash 
    const mountedComponent = mount(<AdminQuestionnairesContainer 
      store={store} 
      isLoading={false} 
      fetchAdminQuestionnaires={fetchAdminQuestionnairesSpy}/>);
    
    expect(toJson(mountedComponent)).toMatchSnapshot();
  });
  
  describe('when questionnaires data is available', () => {  
    const store = mockStore({
      adminQuestionnaires: {
        isLoading: false, 
        data: formattedData,
        errorMessage: null
      },
    });
    
    it('renders self with questionnaires props', () => {   
      const { connectedContainer } = connectedSetup(store); 
      expect(connectedContainer).toHaveProp('questionnaires', formattedData);
      expect(toJson(connectedContainer)).toMatchSnapshot();
    });
    
    it('renders QuestionnairesList component', () => {   
      const isloading = false; 
      const { container } = setup(formattedData, isloading);    

      expect(container).toContainReact(
        <QuestionnairesList questionnaires={formattedData} />
      );
    });
    
    it('does not render Loading component', () => {    
      const { container } = setup(formattedData);       
  
      expect(container).not.toContainReact(<Loading />);
    });
  });
  
  describe('when loading', () => {
    const data = null;
    const isloading = true;     
    const { container } = setup(data, isloading); 
    
    it('renders loading component', () => {  
      expect(container).toContainReact(<Loading />);
    });
    
    it('does not render QuestionnairesList component', () => {  
      expect(container).not.toContainReact(
        <QuestionnairesList questionnaires={formattedData} />
      );
    });
  });
});