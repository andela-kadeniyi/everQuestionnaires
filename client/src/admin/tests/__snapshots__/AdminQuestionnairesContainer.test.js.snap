// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminQuestionnaires container renders without crashing 1`] = `
<AdminQuestionnairesContainer
  fetchAdminQuestionnaires={[Function]}
  isLoading={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Loading>
      <div
        className="loadingScreen"
      >
        ...Loading
      </div>
    </Loading>
  </div>
</AdminQuestionnairesContainer>
`;

exports[`AdminQuestionnaires container when questionnaires data is available renders self with questionnaires props 1`] = `
<AdminQuestionnairesContainer
  fetchAdminQuestionnaires={[Function]}
  isLoading={false}
  questionnaires={
    Object {
      "1": Object {
        "id": 1,
        "title": "Cat breeds",
      },
      "2": Object {
        "id": 2,
        "title": "Cat breeds Questionnaire Part 2",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
